test/FunctionalTests/Microsoft.OData.Client.Tests/DataServiceQueryProviderTests.cs

```diff
@@ -218,11 +298,34 @@ private class Product
 
             public decimal Price { get; set; }
 
+            public Edm.Date LaunchDate { get; set; }
+
             public IEnumerable<string> Comments { get; set; }
 
             public static bool StaticFunction(string parameter) { return true; }
 
             public bool InstanceFunction(string parameter) { return true; }
+
+            [OriginalName("sale")]
+            [UriFunction]
+            public int YearSale(int year) => throw new NotSupportedException();
+        }
+
+        private static class UriFunctions
+        {
+            [OriginalName("now")]
+            [UriFunction]
+            public static DateTimeOffset ServerNow() => throw new NotSupportedException();
+
+            [OriginalName("date")]
+            [UriFunction]
+            public static Edm.Date ServerDate(DateTimeOffset value) => throw new NotSupportedException();
+
+            [UriFunction(true)]

```

Comment:
Is it supported to set `canBeEvaluated` to `true` for instance methods? What is the expected behaviour in that case given that the instance may not exist at the time the query is generated?

Reply:
It is supported, but will only work with resolving on the client if the object is already instantiated on the client.
If we have instance method `bool Test(string)` on client, like this...

```
            [UriFunction(true)]
            public bool Test(string data)
            {
                return string.IsNullOrEmpty(data);
            }
```

Then you can get client resolving with...

```
            Product clientProduct = new Product();
            var localDsc = new DataServiceContext(new Uri("http://root"), ODataProtocolVersion.V4);
            var sut = new DataServiceQueryProvider(localDsc);
            var products = localDsc.CreateQuery<Product>("Products")
                .Where(product => clientProduct.Test(""));

            // Act
            var queryComponents = sut.Translate(products.Expression);

            // Assert
            Assert.Equal(@"http://root/Products?$filter=true", queryComponents.Uri.ToString());
```

Server resolving with...

```
            var localDsc = new DataServiceContext(new Uri("http://root"), ODataProtocolVersion.V4);
            var sut = new DataServiceQueryProvider(localDsc);
            var products = localDsc.CreateQuery<Product>("Products")
                .Where(product => product.Test(product.Name));

            // Act
            var queryComponents = sut.Translate(products.Expression);

            // Assert
            Assert.Equal(@"http://root/Products?$filter=Test($it,Name)", queryComponents.Uri.ToString());
```

And it will of cause refuse to construct query trying to use it like this...

```
            var localDsc = new DataServiceContext(new Uri("http://root"), ODataProtocolVersion.V4);
            var sut = new DataServiceQueryProvider(localDsc);
            var products = localDsc.CreateQuery<Product>("Products")
                .Where(product => product.Test(""));
```

So it is supported, but I don't know how valuable that construct is.
Thanks.